src/main/java/build/buildfarm/instance/stub/StubInstance.java

```diff
@@ -298,25 +299,31 @@ public String getTree(
       int pageSize,
       String pageToken,
       ImmutableList.Builder<Directory> directories) {
-    GetTreeResponse response = contentAddressableStorageBlockingStub
+    Iterator<GetTreeResponse> replies = contentAddressableStorageBlockingStub
         .get()
         .getTree(GetTreeRequest.newBuilder()
             .setInstanceName(getName())
             .setRootDigest(rootDigest)
             .setPageSize(pageSize)
             .setPageToken(pageToken)
             .build());
-    directories.addAll(response.getDirectoriesList());
-    return response.getNextPageToken();
+    // new streaming interface doesn't really fit with what we're trying to do here...
```

Comment:
The effect of the new streaming interface on this is actually less than you think - I'm benefitting here because what used to take multiple calls to getTree with distinct page tokens can now be fed completely, ending on a single empty nextPageToken to complete the sequence. The only call to this method is in worker/operationqueue/Worker.java, where the intended effect was to fetch the entire tree, which this is, as you say, completely in spec for.

Reply:
> what used to take multiple calls to getTree with distinct page tokens can now be fed completely

Yes, that's what I don't see here -- at which point are you replacing a loop of getTree calls with a single getTree call?  The only service implementation I see still calls .onNext() just once once before .onCompleted().

Reply:
The only call to this particular function is
src/test/java/build/buildfarm/server/BuildFarmServerTest.java:314
which is unmodified from the previous implementation because the instance interface hasn't changed (it just manifests as a single call in the case of a successful stream, and expects that a non-empty nextPageToken will be returned for any retriable error).

Gotcha regarding the service implementation, yeah I didn't change that to take advantage. Updating now.